packages/pigeon/lib/generator_tools.dart

```diff
@@ -249,3 +249,42 @@ Map<String, Object> mergeMaps(
   }
   return result;
 }
+
+/// A class name that is enumerated.
+class EnumeratedClass {
+  /// Constructor.
+  EnumeratedClass(this.name, this.enumeration);
+
+  /// The name of the class.
+  final String name;
+
+  /// The enumeration of the class.
+  final int enumeration;
+}
+
+/// Custom codec's custom types are enumerated from 127 down to this number to
+/// avoid collisions with the StandardMessageCodec.
+const int _minimumCodecFieldKey = 20;
+
+/// Given an [Api], return the enumerated classes that must exist in the codec
+/// where the enumeration should be the key used in the buffer.
+Iterable<EnumeratedClass> getCodecClasses(Api api) sync* {
+  final Set<String> names = <String>{};
+  for (final Method method in api.methods) {
+    names.add(method.returnType);
+    names.add(method.argType);
+  }
+  final List<String> sortedNames =
+      names.where((String element) => element != 'void').toList();
+  sortedNames.sort();
+  int enumeration = 127;
+  for (final String name in sortedNames) {
+    if (enumeration < _minimumCodecFieldKey) {
```

Comment:
You know the number of entries before starting the loop; why not just check once outside the loop that there aren't too many?

Reply:
done